### PR TITLE
Streamline `gardener-admission-controller` entrypoint

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR streamlines the main entrypoint of the `gardener-admission-controller` so that it is in line with all other components which have been refactored to native controller-runtime functionality. Similar to #6690.

**Which issue(s) this PR fixes**:
Part of #4251 

**Release note**:

```breaking operator
The `server.https` field of the `gardener-admission-controller` configuration has been renamed to `server.webhooks`. Likewise, the Gardener control plane Helm chart has been changed. Please adapt your `values.yaml` files.
```

### CODE REVIEW

File: pkg/admissioncontroller/apis/config/types.go [hunk 1]
```diff
@@ -44,8 +44,7 @@ type AdmissionControllerConfiguration struct {
 // ServerConfiguration contains details for the HTTP(S) servers.
 type ServerConfiguration struct {
 	// HTTPS is the configuration for the HTTPS webhook server.
```
Comment (plkokanov) [inline pkg/admissioncontroller/apis/config/types.go]:
```suggestion
	// Webhooks is the configuration for the HTTPS webhook server.
```

File: pkg/admissioncontroller/webhooks/add.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	mgr manager.Manager,
+	cfg *config.AdmissionControllerConfiguration,
+) error {
+	var (
+		log         = mgr.GetLogger().WithName("webhook")
+		logSeedAuth = log.WithName(seedauthorizer.AuthorizerName)
+		server      = mgr.GetWebhookServer()
+	)
+
+	graph := seedauthorizergraph.New(mgr.GetLogger().WithName("seed-authorizer-graph"), mgr.GetClient())
```
Comment (plkokanov) [inline pkg/admissioncontroller/webhooks/add.go:47]: Shouldn't this be:
```suggestion
	graph := seedauthorizergraph.New(log.WithName("seed-authorizer-graph"), mgr.GetClient())
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR!
We had a joint review with @ary1992 @toschli and found only some minor stuff.

File: pkg/admissioncontroller/webhooks/add.go (hunk 2 above)
Comment (rfranzke) [reply pkg/admissioncontroller/webhooks/add.go:47]: Initially, I had done it like this, however I found it confusing that the graph logs would be under a "webhook" logger name. The graph itself is no webhook, it's just an "internal tool" used by one of the webhook handler, so I chose to only name it "seed-authorizer-graph".

**Comment by rfranzke** [comment]
Rebased after conflicts, adding labels for merge.

**Comment by rfranzke** [comment]
> @rfranzke: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-gardener-check-vulnerabilities	[d4d3d20](https://github.com/gardener/gardener/commit/d4d3d2096831f3d05883bd52865a31f7b7239163)	[link](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6706/pull-gardener-check-vulnerabilities/1573302582934245376)	false	`/test pull-gardener-check-vulnerabilities`
> pull-gardener-e2e-kind	[d4d3d20](https://github.com/gardener/gardener/commit/d4d3d2096831f3d05883bd52865a31f7b7239163)	[link](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6706/pull-gardener-e2e-kind/1573302583076851712)	true	`/test pull-gardener-e2e-kind`
> [Full PR test history](https://prow.gardener.cloud/pr-history?org=gardener&repo=gardener&pr=6706). [Your PR dashboard](https://prow.gardener.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Arfranzke). [Command help for this repository](https://prow.gardener.cloud/command-help?repo=gardener%2Fgardener). Please help us cut down on flakes by linking this test failure to an [open flake report](https://github.com/gardener/gardener/issues?q=is:issue+is:open+label:kind/flake) or filing a [new flake report](https://github.com/gardener/gardener/issues/new?template=flaking-test.md&title=%5BFlaky+Test%5D+FLAKING+TEST%2FSUITE) if you can't find an existing one. Also see our [testing guideline](https://github.com/gardener/gardener/blob/master/docs/development/testing.md) for how to avoid and hunt flakes.

I think this failed because of a race condition in the seed removal. The gardenlet logs are

```
2022-09-23T14:21:42.994553733Z stderr F panic: failed getting cluster-identity ConfigMap in garden cluster: configmaps "cluster-identity" not found
2022-09-23T14:21:42.99457374Z stderr F 
2022-09-23T14:21:42.994579232Z stderr F goroutine 1 [running]:
2022-09-23T14:21:42.994583529Z stderr F main.main()
2022-09-23T14:21:42.994587285Z stderr F 	github.com/gardener/gardener/cmd/gardenlet/main.go:30 +0x92
```

Probably the configmap was deleted before the gardenlet got restarted (for whatever reason, maybe VPA). After restart, it couldn't start up anymore and fails to properly cleanup the remainings. cc @plkokanov 

/retest

**Comment by rfranzke** [comment]
Again, instabilities in the seed cluster due to VPA eviction. Maybe we should disable it for the e2e tests...

/retest

**Pull request merged**